wdl-grammar/src/v1/lint/one_empty_line.rs

```diff
@@ -0,0 +1,133 @@
+use wdl_core::file::location::Position;
+use wdl_core::file::Location;
+use wdl_core::str::LinesWithOffsetsExt as _;
+use wdl_core::Version;
+
+use crate::v1;
+
+/// At most one empty line in a row.
+#[derive(Debug)]
+pub struct OneEmptyLine;
+
+impl<'a> OneEmptyLine {
+    /// Creates a warning if there is more than one empty line in a row.
+    fn more_than_one_empty_line(&self, location: Location) -> lint::Warning
+    where
+        Self: Rule<&'a Pair<'a, v1::Rule>>,
+    {
+        // SAFETY: this error is written so that it will always unwrap.
+        lint::warning::Builder::default()
+            .code(self.code())
+            .level(lint::Level::Low)
+            .group(self.group())
+            .push_location(location)
+            .subject("more than one empty line")
+            .body("There should be at most one empty line in a row.")
+            .fix("Remove the superfluous empty lines.")
+            .try_build()
+            .unwrap()
+    }
+}
+
+impl<'a> Rule<&'a Pair<'a, v1::Rule>> for OneEmptyLine {
+    fn code(&self) -> Code {
+        // SAFETY: this manually crafted to unwrap successfully every time.
+        Code::try_new(code::Kind::Warning, Version::V1, 11).unwrap()
+    }
+
+    fn group(&self) -> Group {
+        Group::Spacing
+    }
+
+    fn check(&self, tree: &'a Pair<'_, v1::Rule>) -> lint::Result {
+        let mut warnings: VecDeque<_> = VecDeque::new();
+
+        let mut n_empty_lines = 0;
+
+        for (line_no, start_byte_no, _end_byte_no, line) in tree.as_str().lines_with_offsets() {
+            if line.trim().is_empty() {
+                n_empty_lines += 1;
+            } else if n_empty_lines >= 2 {
+                let start =
+                    Position::new(line_no, NonZeroUsize::try_from(1).unwrap(), start_byte_no);
+                warnings.push_back(self.more_than_one_empty_line(Location::Position(start)));
+
+                n_empty_lines = 0;
+            } else {
+                n_empty_lines = 0;
+            }
+        }
+
+        match warnings.pop_front() {
+            Some(front) => {
+                let mut results = NonEmpty::new(front);
+                results.extend(warnings);
+                Ok(Some(results))
+            }
+            None => Ok(None),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use pest::Parser as _;
+    use wdl_core::concern::lint::Rule as _;
+
+    use super::*;
+    use crate::v1::parse::Parser;
+    use crate::v1::Rule;
+
+    #[test]
+    fn it_catches_too_many_empty_lines() -> Result<(), Box<dyn std::error::Error>> {
+        let tree = Parser::parse(
+            Rule::document,
+            r#"version 1.0
+
+
+workflow a_workflow {}"#,
+        )?
+        .next()
+        .unwrap();
+
+        let warnings = OneEmptyLine.check(&tree)?.unwrap();
+        assert_eq!(warnings.len(), 1);
+        assert_eq!(
+            warnings.first().to_string(),
+            "[v1::W011::Spacing/Low] more than one empty line (4:1)"
```

Comment:
This is flagging the next non-whitespace line. Is that what we want? Or should it flag the problematic lines?